pkg/ccl/backupccl/show.go

```diff
@@ -357,36 +358,47 @@ func showBackupPlanHook(
 				KMSInfo: defaultKMSInfo,
 			}
 		}
-		var incPaths []string
-		incStore := store
-		if incDest, ok := opts[backupOptIncStorage]; ok {
-			if subdir != "" {
-				parsed, err := url.Parse(incDest)
-				if err != nil {
-					return err
-				}
-				parsed.Path = path.Join(parsed.Path, subdir)
-				incDest = parsed.String()
-			}
-			incStore, err = p.ExecCfg().DistSQLSrv.ExternalStorageFromURI(ctx, incDest, p.User())
-			if err != nil {
-				return errors.Wrapf(err, "make incremental storage")
-			}
-			defer incStore.Close()
+		explicitIncPaths := make([]string, 0)
+		explicitIncPath := opts[backupOptIncStorage]
+		if len(explicitIncPath) > 0 {
+			explicitIncPaths = append(explicitIncPaths, explicitIncPath)
 		}
-		incPaths, err = FindPriorBackups(ctx, incStore, IncludeManifest)
 
+		collection, computedSubdir := CollectionAndSubdir(dest, subdir)
```

Comment:
When do we need this? If someone does `SHOW BACKUP <collection>/<backup-date>` instead of `SHOW BACKUP <backup-date> IN <collection>` ?

I'm temped to say that with the move to separated per-collection incremental storage, pointing directly to a raw full backup path like the former should be discouraged, or at least, shouldn't try to be magically collection aware, and maybe _just_ shows the full backup at that dest. That implies implying `SHOW BACKUP abc IN col` would show something different (all the incs) than `SHOW BACKUP col/abc` but that seems... maybe okay? i.e. we moved incremtnals to be collection level, and so use the collection syntax to interact with them?

I dunno. Not blocking, just something to chew on / maybe room to remove some code/complexity of supporting both here.

Reply:
Yes, exactly. Aditya caught the regression, discussed in https://cockroachlabs.slack.com/archives/CJXBR3693/p1646276263127099?thread_ts=1646246621.037659&cid=CJXBR3693.

Naively, I don't think I'd expect those two SHOW commands to have different outputs. It seems like a lot of work for a simple `IN` to convey "this is the modern collection syntax," in ways I'd expect to surprise users. But I do agree this code introduces a lot of complexity to get the "it just works" magic, and I would love to think of ways to simplify by reducing functionality.